src/observe-tag.js

```diff
@@ -13,10 +12,23 @@ const byDistanceFromIndex = targetIndex => (indexA, indexB) => {
 	return diffFromTargetA - diffFromTargetB
 }
 
+const hasMatchingTagName = tagName => node => {
+	if (node.tagName === tagName) {
+		return NodeFilter.FILTER_ACCEPT
+	}
+
+	return NodeFilter.FILTER_SKIP
+}
+
 // get an array including the element and all it's children
-const parentAndChildrenElements = node => {
-	const children = node.querySelectorAll('*')
-	return [node, ...children]
+const parentAndChildrenElements = (node, tagName) => {
+	const componentWalker = document.createTreeWalker(node, NodeFilter.SHOW_ELEMENT, hasMatchingTagName(tagName))
+	const parentAndChildren = [componentWalker.currentNode]
+	while (componentWalker.nextNode()) {
+		parentAndChildren.push(componentWalker.currentNode)
+	}
+
+	return parentAndChildren
```

Comment:
Same as above, rather than using `querySelectorAll`, this iterates through all the elements using a `TreeWalker`, and saves those that have the same tagName (since those are the only ones we would reattach focus on).